main.tf

```diff
@@ -103,25 +116,36 @@ resource "azurerm_role_assignment" "key_vault_rbac_current_user" {
   scope                = azurerm_key_vault.KeyVault.id
   role_definition_name = "Key Vault Administrator"
   principal_id         = data.azurerm_client_config.current.object_id
+
+  depends_on = [azurerm_key_vault.KeyVault]
 }
 
 // Create theKey Vault Secrets for later use
 
 resource "azurerm_key_vault_secret" "cd_azure_devops_pat" {
   name         = "cd-azure-devops-pat"
-  value        = ""
+  value        = var.cd_azure_devops_pat_value
   key_vault_id = azurerm_key_vault.KeyVault.id
+
+  depends_on = [azurerm_role_assignment.key_vault_rbac_current_user]
 }
 
 resource "azurerm_key_vault_secret" "cd_github_pat" {
   name         = "cd-github-pat"
-  value        = ""
+  value        = var.cd_github_pat_value
   key_vault_id = azurerm_key_vault.KeyVault.id
+
+  depends_on = [azurerm_role_assignment.key_vault_rbac_current_user]
 }
 
 //Create the provider file for the next module
 
 resource "local_file" "terraform_providers" {
   filename = "${path.root}/../2.0.0.CDMonitoring_Prerequisite_Deployment/providers.tf"
```

Comment:
@cuanl-cd perhaps a templates subdir and add these two files - providers.tf and terraform.tfvars - into there?

Reply:
have added templates directory for template files

Reply:
Does line 144 need to specify it? E.g.

```terraform
filename = "${path.root}/templates/providers.tf"
```